internal/imagemgmt/imagemgmt.go

```diff
@@ -0,0 +1,278 @@
+	GetRemovalCandidates() (removalCandidates []imageMgmtImageInfo, rc error)
+	CleanupUnusedImages(thresholdPercent int) error
+}
+
+type ImageMgmtClient struct {
+	Log         *logr.Logger
+	Executor    ops.Execute
+	StoragePath string
+}
+
+// NewImageMgmtClient creates and returns a ImageMgmtIntf interface for image management commands
+func NewImageMgmtClient(log *logr.Logger, hostCommandsExecutor ops.Execute, storagePath string) ImageMgmtIntf {
+	return &ImageMgmtClient{
+		Log:         log,
+		Executor:    hostCommandsExecutor,
+		StoragePath: storagePath,
+	}
+}
+
+// imageMgmtImageInfo struct provides the image data required for image management
+type imageMgmtImageInfo struct {
+	Id          string
+	RepoTags    []string
+	RepoDigests []string
+	Names       []string
+	Created     uint64
+	Dangling    bool
+}
+
+// Types for image data from crictl commands
+type crictlContainerImageData struct {
+	Image string `json:"image"`
+}
+type crictlContainerData struct {
+	Image crictlContainerImageData `json:"image"`
+}
+
+type crictlContainersList struct {
+	Containers []crictlContainerData `json:"containers"`
+}
+
+type crictlImageData struct {
+	Id     string `json:"id"`
+	Pinned bool   `json:"pinned"`
+}
+
+type crictlImagesList struct {
+	Images []crictlImageData `json:"images"`
+}
+
+// Types for image data from podman commands
+type podmanContainerData struct {
+	ImageID string
+}
+
+// Use a var for syscall.Statfs in order to override it in mock tests
+var syscallStatfs = syscall.Statfs
+
+// CheckDiskUsageAgainstThreshold gets the disk usage for the specified path and checks whether it exceeds the specified threshold
+func (c *ImageMgmtClient) CheckDiskUsageAgainstThreshold(thresholdPercent int) (bool, error) {
+	var stat syscall.Statfs_t
+	if err := syscallStatfs(c.StoragePath, &stat); err != nil {
+		return false, fmt.Errorf("statfs failed for %s: %w", c.StoragePath, err)
+	}
+
+	// Get disk usage percentage
+	usedpct := int(((stat.Blocks - stat.Bavail) * 100) / stat.Blocks)
+	c.Log.Info(fmt.Sprintf("DEBUG: Disk usage of %s is currently %d%%", c.StoragePath, usedpct))
+	return (usedpct > thresholdPercent), nil
+}
+
+// GetInuseImages gets the list of images in use by cri-o or podman containers
+func (c *ImageMgmtClient) GetInuseImages() (inUse []string, rc error) {
+	// Get the list of containers via crictl
+	output, err := c.Executor.Execute("crictl", "ps", "-a", "-o", "json")
+	if err != nil {
+		rc = fmt.Errorf("failed to run crictl ps: %w", err)
+		return
+	}
+
+	var crictlContainers crictlContainersList
+	if err := json.Unmarshal([]byte(output), &crictlContainers); err != nil {
+		rc = fmt.Errorf("unable to parse crictl ps command output: %w", err)
+		return
+	}
+
+	for _, container := range crictlContainers.Containers {
+		inUse = lcautils.AppendToListIfNotExists(inUse, container.Image.Image)
+	}
+
+	// Get the list of containers via podman
+	output, err = c.Executor.Execute("podman", "ps", "-a", "--format", "json")
```

Comment:
why podman? There should not be images running with podman in ocp

Reply:
In which case this is an empty list. But it's possible for there to be podman containers, so why not check?